community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPagedFile.java

```diff
@@ -47,12 +51,12 @@
             UnsafeUtil.getFieldOffset( MuninnPagedFile.class, "lastPageId" );
 
     final MuninnPageCache pageCache;
-    // This is the table where we translate file-page-ids to cache-page-ids:
     final int pageSize;
     final PageCacheTracer tracer;
 
-    final PrimitiveLongObjectMap<MuninnPage>[] translationTables;
-    final StampedLock[] translationTableLocks;
+    // This is the table where we translate file-page-ids to cache-page-ids:
+    volatile Object[][] translationTable;
+    final StampedLock translationTableResizeLock;
```

Comment:
It has to be some lock, though. Would you rather it were a monitor lock on the MuninnPagedFile object?